tests/test_pipeline.py

```diff
@@ -215,6 +215,51 @@ def test_diad_testing_pipeline_output(
     assert "INFO | ~~~ Pipeline finished ~~~" in log_contents
 
 
+@pytest.mark.cupy
+def test_sweep_pipeline_with_save_all(
+    cmd, standard_data, sample_pipelines, standard_loader, output_folder, merge_yamls
+):
+    cmd.insert(7, standard_data)
+    cmd.insert(8, sample_pipelines + "testing/sweep_testing_pipeline.yaml")
+    subprocess.check_output(cmd)
+
+    files = read_folder("output_dir/")
+    assert len(files) == 6
+
+    copied_yaml_path = list(filter(lambda x: ".yaml" in x, files)).pop()
+    with open(sample_pipelines + "testing/sweep_testing_pipeline.yaml", "r") as original_yaml:
+        original_text = original_yaml.read()
+    with open(copied_yaml_path, "r") as copied_yaml:
+        copied_text = copied_yaml.read()
+
+    assert copied_text == original_text
+
+    tif_files = list(filter(lambda x: ".tif" in x, files))
+    assert len(tif_files) == 2
+
+    #: check that the image size is correct
+    imarray = np.array(Image.open(tif_files[0]))
+    assert imarray.shape == (128, 160)
+    assert imarray.sum() == 3855857
+
+    imarray = np.array(Image.open(tif_files[1]))
+    assert imarray.shape == (128, 160)
+    assert imarray.sum() == 3856477
+
+    h5_files = list(filter(lambda x: ".h5" in x, files))
+    assert len(h5_files) == 2
```

Comment:
Given that this test is for ensuring that parameter sweeps run as expected, and the fact that a parameter sweep on a method produces an intermediate hdf5 file with a different structure compared to normal intermediate hdf5 files, would it be worth:
- checking that the expected datasets exist within the hdf5 file
- similar to the check on the tiff files, do a check of the sum and shape of the datasets in the hdf5 file

Reply:
Fair points. I was about to ask about this because when I test the datasets in the hdf5 files, I get an error:

```python
    with h5py.File(h5_files[0], "r") as f:
        print(list(f.keys()))
        print(f["data"].dtype)
```

```console
$ python -m pytest tests/test_pipeline.py::test_sweep_pipeline_with_save_all -rP
                                                                                                         

tests/test_pipeline.py::test_sweep_pipeline_with_save_all FAILED                                                                                            [100%]

============================================================================ FAILURES =============================================================================
________________________________________________________________ test_sweep_pipeline_with_save_all ________________________________________________________________
Traceback (most recent call last):
  File "/scratch/naman/oss/httomo/tests/test_pipeline.py", line 256, in test_sweep_pipeline_with_save_all
    print(f["data"].dtype)
AttributeError: 'Group' object has no attribute 'dtype'
---------------------------------------------------------------------- Captured stdout call -----------------------------------------------------------------------
['data']
```

Reply:
Many thanks for documenting this, Yousef.